samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,197 @@
+    message(STATUS "Please download nanopb from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
+    message(FATAL_ERROR "nanopb required by this project.")
+endif()
+
+message(STATUS "Using protoc: ${PROTOC_BIN}")
+message(STATUS "Using protoc-gen-nanopb: ${NANOPB_BIN}")
+
+if(ALEXA_EMBEDDED_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use Python and west to find the alexa-embedded path
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c
+    "from west.manifest import Manifest; manifest = Manifest.from_file(); print(manifest.get_projects(['Alexa-Gadgets-Embedded-Sample-Code'])[0].posixpath)"
+    OUTPUT_VARIABLE
+    west_project_output
+    RESULT_VARIABLE
+    result
+  )
+  string(REGEX REPLACE "[\r\n]+" "" ALEXA_EMBEDDED_PATH "${west_project_output}")
+  if(${result})
+    message(FATAL_ERROR "Failed to find alexa-embedded, cannot build sample")
+  endif()
+else()
+  message(FATAL_ERROR "west not installed, please provide ALEXA_EMBEDDED_PATH to CMake to build this sample")
+endif()
+
+if(NANOPB_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use Python and west to find the nanopb path
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c
+     "from west.manifest import Manifest; manifest = Manifest.from_file(); print(manifest.get_projects(['nanopb'])[0].posixpath)"
+    OUTPUT_VARIABLE
+    west_project_output
+    RESULT_VARIABLE
+    result
+  )
+  string(REGEX REPLACE "[\r\n]+" "" NANOPB_PATH "${west_project_output}")
+  if(${result})
+    message(FATAL_ERROR "Failed to find nanopb source files, cannot build sample")
+  endif()
+else()
+  message(FATAL_ERROR "west not installed, please provide NANOPB_PATH to CMake to build this sample")
+endif()
+
+function(compile_protobufs_in_dir dir recursive)
+    # Find all .proto files and include directories
+    if (recursive)
+        file(GLOB_RECURSE PROTO_FILES "${dir}/*.proto")
+    else()
+        file(GLOB PROTO_FILES "${dir}/*.proto")
+    endif()
+
+    foreach(file ${PROTO_FILES})
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+    endforeach()
+
+    # Compile .proto files in their respective directories
+    foreach(file ${PROTO_FILES})
+        # Add all directories containing .proto files.
+        # Duplicates and current directory is then removed from "-I" arguments
+        set(PROTOC_ARGS "${PROTOC_INC_DIRS}")
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        get_filename_component(FILE_NAME ${file} NAME)
+        list(REMOVE_ITEM PROTOC_ARGS "-I${FILE_PATH}")
+        list(REMOVE_DUPLICATES PROTOC_ARGS)
+        list(APPEND PROTOC_ARGS "-I.")
+        list(APPEND PROTOC_ARGS "--nanopb_out=${FILE_PATH}")
+        list(APPEND PROTOC_ARGS "${FILE_NAME}")
+        message(STATUS "Compiling proto definition: ${FILE_NAME}")
```

Comment:
This gives excessive log message during CMake execution.
CMake log messages should be kept to a minimum and that are important for the user.
```
-- Using protoc: /projects/github/ncs/bin/nanopb-0.4.2-linux-x86/generator-bin/protoc
-- Using protoc-gen-nanopb: /projects/github/ncs/bin/nanopb-0.4.2-linux-x86/generator-bin/protoc-gen-nanopb
-- Compiling proto definition: alertsDeleteAlertDirective.proto
-- Compiling proto definition: alertsDeleteAlertDirectivePayload.proto
-- Compiling proto definition: alertsSetAlertDirective.proto
-- Compiling proto definition: alertsSetAlertDirectivePayload.proto
-- Compiling proto definition: alexaDiscoveryDiscoverResponseEvent.proto
-- Compiling proto definition: alexaDiscoveryDiscoverResponseEventPayload.proto
-- Compiling proto definition: alexaDiscoveryDiscoverDirective.proto
-- Compiling proto definition: alexaDiscoveryDiscoverDirectivePayload.proto
-- Compiling proto definition: alexaGadgetMusicDataTempoDirective.proto
-- Compiling proto definition: alexaGadgetMusicDataTempoDirectivePayload.proto
-- Compiling proto definition: alexaGadgetSpeechDataSpeechmarksDirective.proto
-- Compiling proto definition: alexaGadgetSpeechDataSpeechmarksDirectivePayload.proto
-- Compiling proto definition: alexaGadgetStateListenerStateUpdateDirective.proto
-- Compiling proto definition: alexaGadgetStateListenerStateUpdateDirectivePayload.proto
-- Compiling proto definition: notificationsClearIndicatorDirective.proto
-- Compiling proto definition: notificationsClearIndicatorDirectivePayload.proto
-- Compiling proto definition: notificationsSetIndicatorDirective.proto
-- Compiling proto definition: notificationsSetIndicatorDirectivePayload.proto
-- Compiling proto definition: directiveHeader.proto
-- Compiling proto definition: eventHeader.proto
-- Compiling proto definition: directiveParser.proto
-- Compiling proto definition: eventParser.proto
-- Compiling proto definition: accessories.proto
-- Compiling proto definition: common.proto
-- Compiling proto definition: device.proto
-- Compiling proto definition: firmware.proto
-- Compiling proto definition: custom_event.proto
```